test-data/unit/check-literal.test

```diff
@@ -2960,3 +2960,90 @@ class C(B[Literal["word"]]):
 reveal_type(C().collection)  # N: Revealed type is "builtins.list[Literal['word']]"
 reveal_type(C().word)  # N: Revealed type is "Literal['word']"
 [builtins fixtures/tuple.pyi]
+
+[case testLiteralAddition]
```

Comment:
Can you add the interesting bits of the inheritance example I gave in an earlier comment?